### DISCUSSION TITLE
[BUG] Creating plots on Windows via CLI sometimes fails on file "permission denied" error.

### DISCUSSION THREAD

**Discussion by KingCubby** [Chia-Network/chia-blockchain#3186] [General]
**Describe the bug**
When creating plots using Windows Powershell or the standard CLI on Windows 10, the plotter will crash with a "Permission denied" error during phase 3 (and sometimes 4)

**To Reproduce**

1. Open Powershell or the Command Prompt on a Windows 10 installation of Chia (I'm using 1.0.3)
2. Start plotting using the CLI (e.g. chia plots create -k 32 -t "temp" -d "farm")
3. Sit and wait for 6-10 hours.

**Expected behavior**
Ideally, I'd end up with a new plot in "farm". Often (about 50% of the time on one machine), I'll get a crash in Phase 3 (once in Phase 4) with a permission denied error on a temp file.

**Screenshots**

(crash dump)
Caught plotting error: Could not open e:\Chia temp\plot-k32-2021-04-12-21-27-f094ef6b400366057257793fd00f1d0b1a3a17320d2334762c2eda48f12b3e37.plot.p2.t4.sort_bucket_101.tmp: Permission denied

Traceback (most recent call last):
  File "src\cmds\chia.py", line 80, in <module>
  File "src\cmds\chia.py", line 76, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 21, in new_func
  File "src\cmds\plots.py", line 134, in create_cmd
  File "src\plotting\create_plots.py", line 176, in create_plots
RuntimeError: Unknown exception
[42620] Failed to execute script chia

**Desktop:**

- OS: Windows 10
- OS Version/Flavor: Version 20H2 / Build 19042.867
- CPU: Intel(R) Core(TM) i9-10900KF CPU @ 3.70GHz   3.70 GHz

**Additional context**

This has happened less often on another machine running an older i7 CPU. It seems to happen more often on my faster SSD being used as a temporary drive, but I'm not certain of this.

**Comment by aqk** [comment]
Wow. This one is unique. At first, I thought it would be the final copy step, and I was going to ask you to investigate the distination directory.

But, your message refers to "e:\Chia temp\plot-k32-2021-04-12-21-27-f094ef6b400366057257793fd00f1d0b1a3a17320d2334762c2eda48f12b3e37.plot.p2.t4.sort_bucket_101.tmp: Permission denied"

Two thoughts:

1. You might have a system service or virus scanner interfering with the plot process
2. Memory corruption. Ease back on your CPU or memory overclock, especially memory. Maybe increase your voltages.

Try to run stable before fast.

Please install https://keybase.io/download
And then come to keybase://chat/chia_network.public#plotting-hardware
for more discussion on plotting.

**Comment by KingCubby** [comment]
After installing a newer version of Chia (I believe my report was for v
1.0.0), and switching to using a drive with more space as a temp folder, I
was more consistently able to plot. I still ran into the occasional
"permission denied" failure during the final delete temp/rename copy step
(mostly the delete part), but as I was easily able to recover from that,
and it was way less consistent, I was able to fully plot my available space
pretty easily.

I'm updated to 1.1 now, and out of spare drive space, so I won't be able to
test further. The virus checker is an interesting hypothesis... Nothing
else is particularly sketchy. This is my main work machine, and it runs
pretty well, so I don't think I was pushing it too hard.

Thanks for the response anyway.

On Tue, Apr 27, 2021 at 5:50 PM Adam Kelly ***@***.***> wrote:

> Wow. This one is unique. At first, I thought it would be the final copy
> step, and I was going to ask you to investigate the distination directory.
>
> But, your message refers to "e:\Chia
> temp\plot-k32-2021-04-12-21-27-f094ef6b400366057257793fd00f1d0b1a3a17320d2334762c2eda48f12b3e37.plot.p2.t4.sort_bucket_101.tmp:
> Permission denied"
>
> Two thoughts:
>
>    1. You might have a system service or virus scanner interfering with
>    the plot process
>    2. Memory corruption. Ease back on your CPU or memory overclock,
>    especially memory. Maybe increase your voltages.
>
> Try to run stable before fast.
>
> Please install https://keybase.io/download
> And then come to keybase://chat/chia_network.public#plotting-hardware
> for more discussion on plotting.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Chia-Network/chia-blockchain/issues/1858#issuecomment-828058386>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD4GRIY4NWM7KL5WSUMGSFLTK5LVBANCNFSM4233WUDQ>
> .
>


-- 
Erik Kiss
Funny Fox Holdings, Inc.
President/Founder
